quint/src/IRprinting.ts

```diff
@@ -181,6 +184,25 @@ export function rowToString(r: Row): string {
   return fields === '' ? '{}' : `{ ${fields} }`
 }
 
+/**
+ * Pretty prints a sum type. Standard sum printing used in error reporting
+ *
+ * @param r the sum type to be formatted
+ *
+ * @returns a string with the pretty printed sum
+ */
+export function sumToString(s: QuintSumType): string {
+  return s.fields.fields
+    .map((f: RowField) => {
+      if (isTheUnit(f.fieldType)) {
+        return `| ${f.fieldName}`
+      } else {
+        return `| ${f.fieldName}(${typeToString(f.fieldType)})`
+      }
+    })
+    .join('\n')
```

Comment:
Do you expect these to get too long to print into a single line?

Otherwise we could just `.join(' | ')?

Reply:
I was following the example of the union printer, but I think you're right that probably the cases for this will be much shorter. I'll make it on a single line.